### PR TITLE
Fix product SKU styles

### LINKED ISSUE #52396: Product SKU looks different than Category and Tags in Single Product template
**Steps to reproduce:**

1. Go to Editor > Single Product template
2. Reset customizations if you've applied any changes

Expected: Product SKU, categories and tags should looks similar as they're used in the same context/way
Actual: Products SKU stands out: font size is bigger and "Suffix" placeholder is present no matter if block is focused or not

Image

For Product Categories and Tags "Suffix" appears only when block is focused:

Image

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR modifies some `Product SKU` styles.

Before

Screenshot 2024-11-14 at 3 24 18 PM

After

Screenshot 2024-11-14 at 3 17 41 PM

Closes #52396.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Editor > Single Product template
2. Reset customizations if you've applied any changes
3. Verify Product SKU looks good

Screenshot 2024-11-14 at 3 20 47 PM

After selecting it the `Suffix` should be visible.

Screenshot 2024-11-14 at 3 17 41 PM

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/sku/block.tsx [hunk 1]
```diff
@@ -44,15 +45,17 @@ const Preview = ( {
...
 			value={ prefix }
 			onChange={ ( value ) => setAttributes( { prefix: value } ) }
 		/>
-		<strong>{ sku }</strong>
+		<span> { sku }</span>
 		<RichText
+			className="wc-block-components-product-sku__suffix"
 			tagName="span"
-			placeholder="Suffix"
+			placeholder={ ' ' + __( 'Suffix', 'woocommerce' ) }
```
Comment (octaedro) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/sku/block.tsx:58]: I added the space here to follow the same pattern that [GB is using](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-terms/edit.js#L149).

**Review by kmanijak — CHANGES_REQUESTED** [review]
Thanks for taking care of it and kudos for adding `i18n`!

I tested the changes and I see some differences between SKU vs Categories/Tags on TT5 theme, SKU is thin, the others are strong:

Place | Screenshot
-- | --
Editor | image
Frontend | image

Also, on TT5 on the frontend the SKU value is "bolder" (font-weight: 700) compared to categories/tags value (font-weight: 600). It's subtle but it caught my attention:

Place | Screenshot
-- | --
Current (font-weight: 700) | image
Changed manually to (font-weight: 600) | image

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductSKU.php [hunk 2]
```diff
@@ -71,18 +71,18 @@ protected function render( $attributes, $content, $block ) {
...
 
 		$suffix = isset( $attributes['suffix'] ) ? wp_kses_post( ( $attributes['suffix'] ) ) : '';
 		if ( ! empty( $suffix ) ) {
-			$suffix = sprintf( '<span class="suffix">%s</span>', $suffix );
+			$suffix = sprintf( '<span class="wp-block-post-terms__suffix">%s</span>', $suffix );
 		}
 
 		return sprintf(
-			'<div class="wc-block-components-product-sku wc-block-grid__product-sku wp-block-woocommerce-product-sku product_meta %1$s" style="%2$s">
+			'<div class="wc-block-components-product-sku wc-block-grid__product-sku wp-block-woocommerce-product-sku product_meta wp-block-post-terms %1$s" style="%2$s">
```
Comment (octaedro) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductSKU.php:83]: To ensure this element mimics the styles of `Category` and `Tags` no matter the theme, I applied to the SKU the class `wp-block-post-terms`.
Same with [prefix](https://github.com/woocommerce/woocommerce/pull/52787/files#diff-df15f52ea37c48bc26b7c89c44cae66ef305373ec59313074fe6819c7eab8d6bR74) and [suffix](https://github.com/woocommerce/woocommerce/pull/52787/files#diff-df15f52ea37c48bc26b7c89c44cae66ef305373ec59313074fe6819c7eab8d6bR79).

**Comment by octaedro** [comment]
Thank you for your feedback, @kmanijak.
I just fixed the problem you mentioned. Could you take another look at this PR?

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductSKU.php (hunk 2 above)
Comment (kmanijak) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductSKU.php:83]: I think that's the right approach! It looks nice in frontend:

image

but I think we should apply the same classes in Editor (screenshot with TT5):

image

Comment (octaedro) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductSKU.php:83]: @kmanijak Done in the commit [c8f4943](https://github.com/woocommerce/woocommerce/pull/52787/commits/c8f494306fa5c9bf82ebfb5b77573b776465e72b)

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/sku/style.scss [hunk 3]
```diff
@@ -3,7 +3,16 @@
 	@include font-size(small);
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/sku/style.scss]: I tested TT5, TT4, TT3 and TT2. On TT2 the size differs.

image

With the current `wp-block-post-terms` class this line is no longer needed as theme takes care of the size (but due to specificity this line takes precedence over theme styles and small font-size is applied).

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/sku/style.scss [hunk 4]
```diff
@@ -3,7 +3,16 @@
...
 	overflow-wrap: break-word;
 
-	strong {
-		text-transform: uppercase;
+	font-size: var(--wp--preset--font-size--small);
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/sku/style.scss]: With the current `wp-block-post-terms` class this line is no longer needed as theme takes care of the size. 🤔

**Review by kmanijak — APPROVED** [review]
I left one last comment. I tested TT5, TT4, TT3 and TT2 with the two marked lines removed and in all of them SKU looks aligned with Categories and Tags. So I'm recommending removing those lines.

Thanks for your patience, pre-approving! 😄

**Pull request merged**